src/fireedge/src/modules/utils/parser/parseServiceTemplate.js

```diff
@@ -0,0 +1,271 @@
+  name,
+  `M|network|${description}||${type}:${value}`,
+]
+
+export const toNetworksValueString = (
+  { name, SIZE: size, type, value },
+  { AR = [], SECURITY_GROUPS = [] } = {}
+) => {
+  if (!name) return
+
+  let extra = []
+
+  if (AR?.length) {
+    const ARs = AR?.map(
+      (ar) =>
+        `AR=[${Object.entries(ar)
+          .map(([key, valueAR]) => `${key}=${valueAR}`)
+          .join(',')}]`
+    )
+
+    extra.push(ARs)
+  }
+
+  if (SECURITY_GROUPS?.length) {
+    const SGs = `SECURITY_GROUPS="${SECURITY_GROUPS.map((sg) => sg?.ID).join(
+      ','
+    )}"`
+
+    extra.push(SGs)
+  }
+
+  if (size) {
+    const SIZE = `SIZE=${size}`
+    extra.push(SIZE)
+  }
+
+  extra = extra?.join(',')
+
+  return {
+    [name]: {
+      [type]: value,
+      extra,
+    },
+  }
+}
+
+export const fromNetworksValueString = (nv) => {
+  const [valueString] = nv
+  const { extra } = valueString
+
+  let conf = []
+
+  const SECURITY_GROUPS = extra
+    ?.match(/SECURITY_GROUPS="([^"]+)"/)
+    ?.pop()
+    ?.split(',')
+    ?.map((id) => ({ ID: id }))
+
+  const AR = extra?.match(/AR=\[([^\]]+)\]/g)?.map((ar) =>
+    Object.fromEntries(
+      ar
+        .replace(/^AR=\[|\]$/g, '')
+        ?.split(',')
+        ?.map((arg) => arg?.split('='))
+    )
+  )
+
+  const SIZE = [
+    extra?.match(/(?:^|,)(SIZE=\d+)(?=,|$)/)?.[1]?.split('=')?.[1],
+  ]?.filter(Boolean)
+
+  if (SECURITY_GROUPS?.length) {
+    conf?.push(['SECURITY_GROUPS', SECURITY_GROUPS])
+  }
+
+  if (AR?.length) {
+    conf?.push(['AR', AR])
+  }
+
+  if (SIZE?.length) {
+    conf?.push(['SIZE', ...SIZE])
+  }
+
+  conf = Object.fromEntries(conf)
+
+  return conf
+}
+
+export const fromNetworkString = (network) => {
+  const [name, networkString] = network
+
+  const [description, , tv] = networkString?.split('|').slice(2)
+  const [type, value] = tv.split(':')
+
+  return {
+    name,
+    description,
+    type,
+    value,
+  }
+}
+
+/**
+ * Recursively converts the keys of an object (and any nested objects) to either lower or upper case.
+ * If the input is an array, it applies the conversion to each item in the array.
+ *
+ * @param {object | Array} obj - The object or array whose keys are to be converted.
+ * @param {boolean} [toLower=true] - Whether to convert keys to lower case. If false, keys are converted to upper case.
+ * @param {number} depth - Control depth of conversion
+ * @returns {object | Array} - The input object or array with keys converted to the specified case.
+ */
+export const convertKeysToCase = (obj, toLower = true, depth = Infinity) => {
+  if (depth < 1) return obj
+
+  if (_.isArray(obj)) {
+    return obj.map((item) => convertKeysToCase(item, toLower, depth))
+  }
+
+  if (_.isObject(obj) && !_.isDate(obj) && !_.isFunction(obj)) {
+    const convertedObj = _.mapKeys(obj, (_value, key) =>
+      toLower ? key.toLowerCase() : key.toUpperCase()
+    )
+
+    if (depth > 1) {
+      return _.mapValues(convertedObj, (value) =>
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).